packages/rollup-contracts/contracts/state-tree/BinaryUtils.sol

```diff
@@ -0,0 +1,199 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+/**
+ MIT License
+ Original author: chriseth
+ */
+
+import {D} from "./DataTypes.sol";
+
+library Utils {
+    /// combines two labels into one
+    function combineLabels(D.Label memory prefix, D.Label memory suffix) internal pure returns (D.Label memory combined) {
+        combined.length = prefix.length + suffix.length;
+        combined.data = prefix.data | (suffix.data >> prefix.length);
+    }
+
+
+    /// Returns a label containing the longest common prefix of `check` and `label`
+    /// and a label consisting of the remaining part of `label`.
+    function splitCommonPrefix(D.Label memory label, D.Label memory check) internal pure returns (D.Label memory prefix, D.Label memory labelSuffix) {
+        return splitAt(label, commonPrefix(check, label));
+    }
+    /// Splits the label at the given position and returns prefix and suffix,
+    /// i.e. prefix.length == pos and prefix.data . suffix.data == l.data.
+    function splitAt(D.Label memory l, uint pos) internal pure returns (D.Label memory prefix, D.Label memory suffix) {
+        require(pos <= l.length && pos <= 256, "Bad pos");
+        prefix.length = pos;
+        if (pos == 0) {
+            prefix.data = bytes32(0);
+        } else {
+            prefix.data = l.data & ~bytes32((uint(1) << (256 - pos)) - 1);
+        }
+        suffix.length = l.length - pos;
+        suffix.data = l.data << pos;
+    }
+    /// Returns the length of the longest common prefix of the two labels.
+    function commonPrefix(D.Label memory a, D.Label memory b) internal pure returns (uint prefix) {
```

Comment:
Would update this to be `commonPrefixLength(...)` because it's not actually returning the common prefix.